pkg/terraform/resource/store.go

```diff
@@ -0,0 +1,43 @@
+package resource
+
+// ConfigStore stores a global configuration
+type ConfigStore struct {
+	config map[string]*Configuration
+}
```

Comment:
```suggestion
type ConfigStore map[string]Configuration
```

If we do that, we can use the pattern of `if cfg, ok := configMap[resourceName]; ok` easily. `nil` check is also possible but that could be simpler to work with.

Reply:
I believe we will end up with more configurations here that are not resource-specific (e.g. provider-specific). So, I would prefer something like the following to leave some room for future configurations.

```
type ConfigStore struct {
	resourceConfig map[string]Configuration
}
```

Reply:
I did some refactoring and moved configuration to a dedicated package (config), which is also a preparation for #61 where developers could provide configurations like which groups to generate, which resources to skip as part of this.